src/cscapi/client.py

```diff
@@ -85,23 +86,48 @@ def add_signals(self, signals: List[SignalModel]):
         for signal in signals:
             self.storage.update_or_create_signal(signal)
 
-    def prune_failing_machines_signals(self):
-        signals = self.storage.get_all_signals()
-        for machine_id, signals in _group_signals_by_machine_id(signals).items():
-            machine = self.storage.get_machine_by_id(machine_id)
-            if machine.is_failing:
-                self.storage.delete_signals(signals)
+    def prune_failing_machines_signals(
+        self, batch_size: int = SIGNAL_BATCH_LIMIT
+    ) -> int:
+        total_pruned = 0
+        while True:
+            signals = self.storage.get_signals(limit=batch_size, is_failing=True)
+            if not signals:
+                break
 
-    def send_signals(self, prune_after_send: bool = True):
-        unsent_signals_by_machineid = _group_signals_by_machine_id(
-            filter(lambda signal: not signal.sent, self.storage.get_all_signals())
-        )
-        self._send_signals_by_machine_id(unsent_signals_by_machineid, prune_after_send)
+            signal_ids = [signal.alert_id for signal in signals]
+            self.storage.delete_signals(signal_ids)
+            total_pruned += len(signals)
+
+        self.logger.info(f"Total pruned signals: {total_pruned}")
+        return total_pruned
+
+    def send_signals(
+        self, prune_after_send: bool = True, batch_size: int = SIGNAL_BATCH_LIMIT
+    ) -> int:
+        offset = 0
+        total_sent = 0
+        while True:
+            signals = self.storage.get_signals(
+                limit=batch_size, offset=offset, sent=False, is_failing=False
+            )
+            if not signals:
+                self.logger.info(f"No signals to send, stopping sending")
+                break
+            unsent_signals_by_machineid = _group_signals_by_machine_id(signals)
+
+            batch_sent = self._send_signals_by_machine_id(
+                unsent_signals_by_machineid, prune_after_send
+            )
+            total_sent += batch_sent
```

Comment:
[warn] tu incrémentes pas offset pour le cas d'un prune=false ?

Reply:
it's ok as we are taking only Sent=False signals